pywbem_mock/_cimrepository.py

```diff
@@ -128,7 +132,7 @@ class defining the data store for the CIM repository.
         # this list is still open.
         self.enumeration_contexts = {}
 
-        # Current namespace.  Set by the mock calls it is used by the
+        # Current namespace.  Set by the caller it is used by the
         # request methods to determine the current namespace.  This is used
         # because namespace is NOT part of the WBEMConnection calls.
         #
```

Comment:
This comment is no longer true when the namespace is used from the parameters. It's function is now only to serve as a default namespace if the operation does not specify one (which is the semantics under which it was passed in).

Reply:
Actually I believe that what I was looking for was to force an explicit namespace definition for each operation. First, there really is no concept of default namespace on the server side, just the namespace that this operation uses and that should be explicit in each operation.  That makes the server operations slightly different than the client operations but does clarify the whole issue for the server.  Any ambiguities associated with the client interface and the imethodcall to mock use of namespace we should sort out in _Fake_wbemconnection so that each CIMRepository method 1)requires an explicit namespace and uses that namespace.  If some method need to verify that the instancename namespace and explicit namespace input parameter match, that they should do.

Reply:
Ok, you're right in that the server does not have a default namespace.

This means, the `default_namespace` init argument of class `CIMRepository` should be removed, and also its `namespace` attribute. The operation methods of `CIMRepository` should require their namespace arguments.

Reply:
Yes.  Also we make namespace positional and the first position in all methods which differs from the client interface. In addition, we should sort out the issue of CIMClassName vs. string representation for classname before the call to the CIM repository.

There goes today and tomorrow.

Started this a while ago. There goes today

Reply:
I would prefer CIMClassName for operations that operate on a single class object, instead of a combination of namespace and class name

Reply:
Agreed That has been removed. DONE.   Each MainProvider responder method now recieves the namespace argument as the first argument (to be consistent)

Reply:
NOTE: All of this conversation has been resolved when we made namespace a required positional parameter on all of the request methods in what is now MainProvider